### PR TITLE
[FLINK-26508] Webhook should only validate on /validate endpoint end log errors for others

### PR DESCRIPTION
**Pull request by SteNicholas**
The current webhook implementations accept requests on all paths and execute validations, which should restrict this to only /validate and log errors on other rest paths.

**The brief change log**
- `AdmissionHandler` adds the check whether the request path is equal to `/validate` and the log of the error on other rest paths.

### CODE REVIEW

File: flink-kubernetes-webhook/src/main/java/org/apache/flink/kubernetes/operator/admission/AdmissionHandler.java [hunk 1]
```diff
@@ -66,15 +70,26 @@ public AdmissionHandler(Validator<GenericKubernetesResource> validator) {
...
+                review = objectMapper.readValue(in, AdmissionReview.class);
+                AdmissionReview response = validatingController.handle(review);
+                sendResponse(ctx, objectMapper.writeValueAsString(response));
+            } catch (Exception e) {
+                sendError(ctx, ExceptionUtils.getStackTrace(e));
+            }
+        } else {
+            String error =
+                    String.format(
+                            "Webhook accepts the illegal request path '%s', only accepts the request path '/validate'.",
```
Comment (gyfora) [inline flink-kubernetes-webhook/src/main/java/org/apache/flink/kubernetes/operator/admission/AdmissionHandler.java:89]: Could we please change this to: 
```
Illegal path requested: '%s'. Only /validate is accepted.
```

**Review by gyfora — CHANGES_REQUESTED** [review]
Looks good, I had a small comment to this. 

@SteNicholas would you have time adding a simple test for this ? That would also cover https://issues.apache.org/jira/browse/FLINK-26137 . Two birds, one stone :)

**Comment by SteNicholas** [comment]
@gyfora ,  I would like to create webhook REST api test in this pull request later.

File: flink-kubernetes-webhook/src/main/java/org/apache/flink/kubernetes/operator/admission/AdmissionHandler.java (hunk 1 above)
Comment (SteNicholas) [reply flink-kubernetes-webhook/src/main/java/org/apache/flink/kubernetes/operator/admission/AdmissionHandler.java:89]: @gyfora , of course yes. The above error message looks good to me.

File: flink-kubernetes-webhook/src/main/java/org/apache/flink/kubernetes/operator/admission/AdmissionHandler.java [hunk 2]
```diff
@@ -66,15 +70,25 @@ public AdmissionHandler(Validator<GenericKubernetesResource> validator) {
...
-        try {
-            InputStream in = new ByteBufInputStream(msgContent);
-            review = objectMapper.readValue(in, AdmissionReview.class);
-            AdmissionReview response = validatingController.handle(review);
-            sendResponse(ctx, objectMapper.writeValueAsString(response));
-        } catch (Exception e) {
-            sendError(ctx, ExceptionUtils.getStackTrace(e));
+        QueryStringDecoder decoder = new QueryStringDecoder(httpRequest.uri());
+        String path = decoder.path();
+        if ("/validate".equals(path)) {
```
Comment (Aitozi) [inline flink-kubernetes-webhook/src/main/java/org/apache/flink/kubernetes/operator/admission/AdmissionHandler.java:75]: nit: could make `/validate` static and reuse it in the error msg :)

Comment (Aitozi) [inline flink-kubernetes-webhook/src/main/java/org/apache/flink/kubernetes/operator/admission/AdmissionHandler.java:75]: nit: could make `/validate` static and reuse it in the error msg :)

**Comment by SteNicholas** [comment]
@gyfora , I will push a commit for the Rest test for `AdmissionReview`. Does I need to push another pull request for the test?

**Pull request merged**